integration/control_plane.go

```diff
@@ -33,33 +38,24 @@ func (f *ControlPlane) Start() error {
 		return err
 	}
 
-	if f.ControllerManager == nil {
-		f.ControllerManager = &ControllerManager{}
-	}
-	f.ControllerManager.APIServerURL = f.APIServer.URL
-	if err := f.ControllerManager.Start(); err != nil {
-		return err
+	for _, c := range f.AdditionalComponents {
+		c.RegisterTo(f)
+		if err := c.Start(); err != nil {
+			return err
+		}
 	}
 
-	if f.Scheduler == nil {
-		f.Scheduler = &Scheduler{}
-	}
-	f.Scheduler.APIServerURL = f.APIServer.URL
-	return f.Scheduler.Start()
+	return nil
 }
 
 // Stop will stop your control plane processes, and clean up their data.
 func (f *ControlPlane) Stop() error {
-	if f.Scheduler != nil {
-		if err := f.Scheduler.Stop(); err != nil {
-			return err
-		}
-	}
-	if f.ControllerManager != nil {
-		if err := f.ControllerManager.Stop(); err != nil {
-			return err
+	for _, c := range f.AdditionalComponents {
```

Comment:
you used to stop them in opposite order, not anymore?

Reply:
- When we stick with the `AdditionalComponents` slice, we cannot assume any order, neither on `Start` nor on `Stop`.
- The only thing we try to guarantee is, that the APIServer is up (and by that, Etcd is up) before we start any of those other comments. The order of the `AdditionalComponents` shouldn't matter, they only have a dependency on APIServer.
- We might even think of starting/stopping them in parallel -- but that's for a different commit/PR.